### PR TITLE
feat(genesis): deployerAddress passed as parameter

### PR DESCRIPTION
**Pull request by Villaquiranm**
closes https://github.com/gnolang/gno/issues/2573

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
</details>

### CODE REVIEW

**Comment by Villaquiranm** [comment]
closes #2573

**Review by moul — CHANGES_REQUESTED** [review]
Depends on https://github.com/gnolang/gno/issues/2824

**Comment by Villaquiranm** [comment]
depends on #3041

**Comment by thehowl** [comment]
No longer blocked; please merge master to fix the go.mod CI and fix lint CI. I'll wait for the review team to do a first pass.

File: gno.land/cmd/gnoland/start.go
Comment (n0izn0iz) [inline gno.land/cmd/gnoland/start.go:400]: the first tx is not necessarily what has been passed to `TxsAddPackages` command

also loading the examples here will be removed from what I understand so maybe we should not change his file until https://github.com/gnolang/gno/issues/1952 is fixed

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: gno.land/cmd/gnoland/start.go
Comment (Villaquiranm) [reply gno.land/cmd/gnoland/start.go]: thanks for the review @n0izn0iz changed back while waiting for that change :)

**Comment by jefft0** [comment]
Hello @Villaquiranm . This PR closes issue https://github.com/gnolang/gno/issues/2573 . I see that it adds a way to configure the deployer address from the command line. But the issue also says "". But the issue also says to change to a default "deployer address like `g1sh0uldn0thappen...`" Is this still needed?

**Comment by Villaquiranm** [comment]
Hey @jefft0 thanks for your review.
I mean I could put g1sh0uldn0thappen but that being a non valid address, maybe it is better to fail if the parameter is not given ? that's why I left it as it was but sure I could put 'g1sh0uldn0thappen'

> Hello @Villaquiranm . This PR closes issue #2573 . I see that it adds a way to configure the deployer address from the command line. But the issue also says "". But the issue also says to change to a default "deployer address like `g1sh0uldn0thappen...`" Is this still needed?

**Comment by jefft0** [comment]
> Hey @jefft0 thanks for your review. I mean I could put g1sh0uldn0thappen but that being a non valid address, maybe it is better to fail if the parameter is not given ? that's why I left it as it was but sure I could put 'g1sh0uldn0thappen'
> 
> > Hello @Villaquiranm . This PR closes issue #2573 . I see that it adds a way to configure the deployer address from the command line. But the issue also says "". But the issue also says to change to a default "deployer address like `g1sh0uldn0thappen...`" Is this still needed?

Hi @Villaquiranm . I can't tell you the answer. My job as part of "review/triage" is to anticipate the questions that the core devs would ask during their final review, and to make sure they have been considered.

The mnemonic (and private key) of address g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5 is well-known, and thehowl suggests that it shouldn't be the default (like it is in your PR). What do you think are the options to deal with this?

**Comment by jefft0** [comment]
Removed the `review/triage-pending` label because zivkovicmilos changed this PR to draft status.

**Pull request closed without merging**
